File: src/diaryApp/Diaries.java
package diaryApp;

import java.util.ArrayList;

public class Diaries {
    private ArrayList<Diary> diaries = new ArrayList<>();
    private boolean isLocked = false;


    public void addDiary(String userName, String passWord) {
        Diary newDiary = new Diary(userName, passWord);
        diaries.add(newDiary);
    }


    public void delete(String username, String password) {
        Diary diary = findIdByUsername(username);
        if (diary != null && diary.validatePassword(password)) {
            diaries.remove(diary);
        }else{
            throw new IllegalArgumentException("Invalid username or password");
        }
    }


    public Diary findIdByUsername(String username) {
        for(Diary diary : diaries) {
            if(diary.getUsername().equals(username)) {
                return diary;
            }
        }
        return null;
    }


    public boolean checkPassword(String username, String password) {
        Diary diary = findIdByUsername(username);
        return diary != null && diary.validatePassword(password);
    }





}

File: ExtraWork/diaryapp/diaries.py
from ExtraWork.diaryapp.diary import Diary

class Diaries:
    def __init__(self):
        self.data = []

    def add_diary(self, username, password):
        new_diary = Diary(username, password)
        self.data.append(new_diary)

    def delete(self, username, password):
        diary = self.find_username(username)
        if diary is not None and diary.validate_password(password):
            self.data.remove(diary)
        else:
            raise ValueError("Invalid username or password")

    def find_username(self, username):
        for diary in self.data:
            if diary.username == username:
                return diary
        return None

